implementations/micrometer-registry-opentelemetry/src/main/java/io/micrometer/opentelemetry/OpenTelemetryRegistry.java

```diff
@@ -0,0 +1,156 @@
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.internal.DefaultMeter;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.metrics.AsynchronousInstrument;
+import io.opentelemetry.metrics.DoubleCounter;
+import io.opentelemetry.metrics.DoubleSumObserver;
+import io.opentelemetry.metrics.DoubleValueObserver;
+import io.opentelemetry.metrics.DoubleValueRecorder;
+import io.opentelemetry.metrics.SynchronousInstrument;
+
+
+/**
+ * A registry for exporting Micrometer Metrics to an OpenTelemetry registry
+ *
+ * @author Erin Schnabel
+ */
+public class OpenTelemetryRegistry extends MeterRegistry {
+
+    final io.opentelemetry.metrics.Meter otelMeter;
+
+    public OpenTelemetryRegistry(OpenTelemetryConfig config, Clock clock) {
+        super(clock);
+        config.requireValid();
+        config().onMeterRemoved(this::onMeterRemoved);
+        this.otelMeter = OpenTelemetry.getMeter(config.instrumentationName(), config.instrumentationVersion());
+    }
+
+    @Override
+    protected <T> Gauge newGauge(Meter.Id id, T obj, ToDoubleFunction<T> valueFunction) {
+        DoubleValueObserver.Builder builder = otelMeter.doubleValueObserverBuilder(id.getName());
+        copyAttributes(id, builder);
+        if ( id.getBaseUnit() != null ) {
+            builder.setUnit(id.getBaseUnit());
+        }
+        return new OpenTelemetryGauge<T>(id, obj, valueFunction, builder.build());
+    }
+
+    @Override
+    protected Counter newCounter(Meter.Id id) {
+        DoubleCounter.Builder builder = otelMeter.doubleCounterBuilder(id.getName());
+        copyAttributes(id, builder);
+        if ( id.getBaseUnit() != null ) {
+            builder.setUnit(id.getBaseUnit());
+        }
+        return new OpenTelemetryCounter(id, builder.build());
+    }
+
+    @Override
+    protected Timer newTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig, PauseDetector pauseDetector) {
+        DoubleValueRecorder.Builder builder = otelMeter.doubleValueRecorderBuilder(id.getName())
+                .setUnit("s");
+        copyAttributes(id, builder);
+        return new OpenTelemetryTimer(id, clock, builder.build(), TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected DistributionSummary newDistributionSummary(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig, double scale) {
+        return null;
```

Comment:
I think this would end up being mapped to a ValueRecorder. At the moment, we'd need to ignore the config, but hopefully when we get a Views API, that could be used to configure a View that would match.

Just FYI, there's has been chatter (maybe just in my head?) about the possibility of views  that are backed by micrometer implementations. In that case, you'd use the OTel APIs, but provide micrometer-implemented views to provide functionality that might be missing from OTel.

Speaking of which, did we ever end up doing the experiment that @kenfinnigan was working on, where the OTel Meter API would be implemented with a micrometer-backed implementation?

Reply:
He has it started. We're doing a little in parallel.

IMO: The two APIs will both continue to exist and be used.. so it's all about what happens behind them to make everything smooth.

Reply:
@jkwatson it's in progress, getting back into it after a break. Putting together an application to exercise all the aspects to validate it all works

Reply:
It would be cool if we can use the same app with this registry, to compare and contrast the outputs.

Reply:
I think that would work, as I was planning on having a mix of usage with Micrometer and OTeL APIs to ensure the two work seamlessly